company-lsp.el

```diff
@@ -451,7 +451,6 @@ See the documentation of `company-backends' for COMMAND and ARG."
     (prefix (and
              (bound-and-true-p lsp-mode)
              (lsp--capability "completionProvider")
-             (not (company-in-string-or-comment))
```

Comment:
I'm afraid this has to be a per-server configuration. Not all language servers handle comments and strings well.